app/src/main/resources/script.js

```diff
@@ -1,38 +1,26 @@
-function shippingCb(address, tokenCallback) {
-    tokenCallback({ // Can return price based on address
-        shippingMethods: [
-            {
-                id: '0',
-                name: 'Standard Ground (5-9 business days)',
-                deliveryTime: '5 - 9 Business Days',
-                cost: 0
-            },
-        ],
-        tax: 0
-    });
+function initiatePayment() {
+    var XHR = new XMLHttpRequest();
+
+    // Set up our request
+    XHR.open('POST', 'http://localhost:3000/transfer', true);
+
+    XHR.setRequestHeader("Content-Type", "application/json; charset=utf-8");
+
+    var data = $.param({
+        merchantId: 'Merchant 123',
+        amount: 4.99,
+        currency: 'EUR',
+        description: 'Book Purchase',
+        destination: 'DE16700222000072880129'
+     });
+
+     // Define what happens on successful data submission
+     XHR.addEventListener("load", function(event) {
+       window.location.replace(event.target.responseURL);
```

Comment:
you are using the redirect flow here, but you can keep working with the popup, check merchant-demo code: https://github.com/tokenio/merchant-demo/blob/master/src/components/TokenEnablerButton/index.js#L74 it can take the token request url instead of terms, let me know if you need my help :-)

Reply:
I think we want to demonstrate plain redirect since that's what our documentation shows. I originally tried using the bindButton set up but it caused all sorts of problems and confusion, so figured for the sample we're giving to devs we should keep it as plain as can be, and show that all Token specific stuff can be kept to the backend 😀